### PR TITLE
Bug 1160992 - Disable the data connection toggle if no sim card r=gasolin

### PR DESCRIPTION
**Pull request by evanxd**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e894ec7ae88408ca9d7246e58e7c97046ba1cac7

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=751d7b3ad1a4caeb0bb9afee41785a6455a08dba

File: apps/settings/js/carrier.js [hunk 1]
```diff
@@ -119,6 +119,9 @@ var CarrierSettings = (function() {
       });
     }
 
+    if (!_mobileConnections[0].iccId) {
+      dataToggle.disabled = true;
```
Comment (gasolin) [inline apps/settings/js/carrier.js]: please add unit test case to make sure it works in future change

Comment (evanxd) [inline apps/settings/js/carrier.js]: Sure. I'll do that if the change is on the right way.

File: apps/settings/js/carrier.js [hunk 2]
```diff
@@ -119,6 +119,9 @@ var CarrierSettings = (function() {
       });
     }
 
+    if (!_mobileConnections[0].iccId) {
```
Comment (gasolin) [inline apps/settings/js/carrier.js]: Not sure if its sufficient for dual sim case. Have you test that in flame?

Comment (evanxd) [inline apps/settings/js/carrier.js]: I've tested in flame. But yes, we need do handle multiple sim cards case here. Thanks.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=10f4b649c20def617ce09674eee3192cbe0a9c2f

File: apps/settings/js/carrier.js [hunk 3]
```diff
@@ -119,6 +119,15 @@ var CarrierSettings = (function() {
       });
     }
 
+    var isThereNoSimCard = Array.prototype.every.call(
```
Comment (gasolin) [inline apps/settings/js/carrier.js]: please add some comment to explain what the intention is, such as `//Disable the data connection toggle if no sim card`

Comment (evelynhung) [inline apps/settings/js/carrier.js]: nit:  change `isThereNoSimCard` to `isSimCardAbsent`

File: apps/settings/js/carrier.js [hunk 4]
```diff
@@ -119,6 +119,15 @@ var CarrierSettings = (function() {
...
+      navigator.mozMobileConnections,
+      function(connection) {
+        return !connection.iccId;
+      }
+    );
+    if (isThereNoSimCard) {
+      dataToggle.disabled = true;
```
Comment (evelynhung) [inline apps/settings/js/carrier.js]: Should we turn the toggle off before disabling it? It looks weird to me if the toggle is on but disabled.

File: apps/settings/js/carrier.js (hunk 3 above)
Comment (evanxd) [inline apps/settings/js/carrier.js]: Sure.

File: apps/settings/js/carrier.js (hunk 4 above)
Comment (evanxd) [inline apps/settings/js/carrier.js]: Sure, let's toggle off first.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e5dcae361f67185ea1f386ffa017e18742b8d849

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=96cac20ba10064c7f2b2973067191137ec6785dc

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=efc65cecfc733376e7c1a3122d2a25c1d533d0d4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=695e0485b8e0d15be65140c7ae645d8f4fbe56af

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=eb64215108ca533c79b61b6fa9c9530a25053165

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2f54dcf2ed253c87cf3aca6b45a0e67af58d75ed

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d2f630909fcb0b76a7ba86c3bb987ed23a1d249d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8c3a5e4deef12af99e4c46d447d09e5401b50e90

**Pull request merged**